src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -101,11 +101,6 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
 
         // Be aware of if we need to re-look for kernels on extension change
         this.extensions.onDidChange(this.onDidChangeExtensions, this, this.disposables);
-
-        if (this.isLocalLaunch) {
-            // Pre-warm fetching local kernels, for remote connections fetch as and when needed.
-            this.getNotebookControllers().catch(traceError);
```

Comment:
Not sure how this slows things down? Is it the python extension activation?

Reply:
I guess it feels like we need to do this so that running a cell doesn't take forever. I'm wondering how this slows down the open as it should be a promise that's going to at least going to go off and look on disk for kernelspecs.

Reply:
Yes,
This causes pyton extension actiavtion, hence slows down extenison host and opening of the notebook.
We alraedy have code that gets this information after we get the notebook opened event.

Reply:
> t should be a promise that's going to at least going to go off and look on disk for kern

Python extnesion spins up processes (process spawning is a sync process), all of this and other code (such as loading python extension js is synchronous), that blocks the extenssion host hence slows the current process of opening a notebook.